.travis.yml

```diff
@@ -26,6 +27,10 @@ jobs:
   include:
     - stage: build
       name: Build and test striot
+      script:
+        - cabal configure --enable-tests && cabal build && cabal test
+        - ./gen_test_makefile.sh > Makefile
+        - make
```

Comment:
If/once #89 is merged, this line could be `make GHC=ghc` for Travis.

Reply:
We unfortunately need to run gen_test_makefile and make *twice*:

 * first `./gen_test_makefile.sh` sets up the top-level Makefile, discovering what examples there are
 * first `make` builds all the `generate`, and runs them
 * once the `generate` has run, there are node? sub-directories containing node.hs files
 * second `./gen_test_makefile.sh` run discovers those node.hs files and generates rules to compile them 

I can probably improve this in the future

Reply:
> - first ./gen_test_makefile.sh sets up the top-level Makefile, discovering what examples there are
>- first make builds all the generate, and runs them
>- once the generate has run, there are node? sub-directories containing node.hs files
>- second ./gen_test_makefile.sh run discovers those node.hs files and generates rules to compile them

My fault here, I saw this behaviour described in a comment from the recent PR updating `gen_test_makefile.sh` but it didn't quite click. I'll make the updates once #89 is merged!

Reply:
I've made the relevant updates and it looks like we're able to build all the examples successfully now!